tpls/desul/include/desul/atomics/Compare_Exchange_OpenMP.hpp

```diff
@@ -18,6 +18,11 @@ SPDX-License-Identifier: (BSD-3-Clause)
 namespace desul {
 namespace Impl {
 
+template <class T>
+constexpr bool host_atomic_is_always_lock_free() {
+  return __atomic_always_lock_free(sizeof(T),0);
```

Comment:
Is it safe to assume that the compiler for `OpenMPTarget` always defines this function? I guess we only allow `clang` which should define it(?).

Reply:
Clang does ... but it also looks like it is just wrong for e.g. a `const atomic<__int128>` on some archs ... https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110061
I will revert all the host side to only 4 and 8 byte atomics.